velero-plugin-for-microsoft-azure/common.go

```diff
@@ -30,21 +30,41 @@ const (
 	cloudNameEnvVar      = "AZURE_CLOUD_NAME"
 
 	resourceGroupConfigKey = "resourceGroup"
+	credentialsFileKey     = "credentialsFile"
 )
 
-func loadEnv() error {
-	envFile := os.Getenv("AZURE_CREDENTIALS_FILE")
-	if envFile == "" {
+func loadEnvFromCredentialsFile(config map[string]string) error {
+	// Prioritize the credentials file path in config, if it exists
+	credentialsFile, ok := config[credentialsFileKey]
+	if ok {
+		// Check that the provided credentialsFile exists on disk
+		if _, err := os.Stat(credentialsFile); err != nil {
+			if os.IsNotExist(err) {
+				return errors.Wrapf(err, "provided credentialsFile does not exist")
+			}
+			return errors.Wrapf(err, "could not get credentialsFile info")
+		}
+	} else {
+		// If a credentials file does not exist in the config, fall back to
+		// any credentials file specified in the environment.
+		credentialsFile = os.Getenv("AZURE_CREDENTIALS_FILE")
+	}
+
+	if credentialsFile == "" {
 		return nil
 	}
 
-	if err := godotenv.Overload(envFile); err != nil {
-		return errors.Wrapf(err, "error loading environment from AZURE_CREDENTIALS_FILE (%s)", envFile)
+	if err := godotenv.Overload(credentialsFile); err != nil {
+		return errors.Wrapf(err, "error loading environment from credentials file (%s)", credentialsFile)
 	}
 
 	return nil
 }
 
+func loadEnv() error {
```

Comment:
Since this function is reduced to just this one line, maybe it's better to get rid of it and call `loadEnvFromCredentialsFile(map[string]string{})` directly both in the object_store.go and in the volume_snapshotter.go files. What do you think? I don't think having it adds any benefit to the code.

Reply:
Yeah, good thinking :+1: I think I initially put this in to avoid changes to the volume snapshotter file, but you're right that it doesn't really offer any benefit.